src/parser/mod.rs

```diff
@@ -10508,6 +10508,43 @@ impl<'a> Parser<'a> {
             }
             rewrite_projection
         } else {
+            // `align` not found, report an error if the projection contains `range_fn` but without rewriting them.
+
+            let align_fill_validate = |expr: &Expr| {
+                rewrite_calculation_expr(expr, true, &mut |e: &Expr| match e {
+                    Expr::Function(func) => {
+                        if let Some(name) = func.name.0.first() {
+                            if name.value.as_str() == "range_fn" {
+                                let FunctionArguments::List(args) = &func.args else {
+                                    unreachable!()
+                                };
```

Comment:
Can we ensure this is unreachable? If not, can we return an error here?

Reply:
Yes, we can.

Reply:
We may use `if-let` or add a safety note that mentions which place ensures this.